lisa/tests/base.py

```diff
@@ -462,8 +472,35 @@ def test_noisy_tasks(self, noise_threshold_pct=None, noise_threshold_ms=None):
             threshold_s = min(threshold_s, noise_threshold_ms * 1e3)
 
         df = self.trace.analysis.tasks.df_tasks_runtime()
-        test_tasks = list(map(self.trace.get_task_pid, self.rtapp_profile.keys()))
-        df_noise = df[~df.index.isin(test_tasks + self.NOISE_IGNORED_PIDS)]
+
+        # We don't want to account the test tasks
+        ignored_pids = list(map(self.trace.get_task_pid, self.rtapp_profile.keys()))
+
+        def compute_duration_pct(row):
+            return row.runtime * 100 / self.trace.time_range
+
+        df["runtime_pct"] = df.apply(compute_duration_pct, axis=1)
+
+        # Figure out which PIDs to exclude from the thresholds
+        for key, threshold in self.NOISE_ACCOUNTING_THRESHOLDS.items():
+            # Find out which task(s) this threshold is about
+            if isinstance(key, int):
+                pids = [key]
+            elif isinstance(key, str):
+                comms = [comm for comm in df.comm.values if re.match(key, comm)]
+                pids = df[df.comm.isin(comms)].index.values
+            else:
+                pids = []
+
+            # For those tasks, check the threshold
+            ignored_pids += [pid for pid in pids if df.loc[pid].runtime_pct <= threshold]
+
+        log_pids = ["{} ({})".format(pid, df.loc[pid].comm) for pid in ignored_pids]
+        self.get_logger().info(
+            "Ignored PIDs for noise contribution: %s", ", ".join(log_pids))
+
+        # Filter out unwanted tasks (rt-app tasks + thresholds)
+        df_noise = df[~df.index.isin(ignored_pids)]
```

Comment:
I can't comment on the other lines not touched here, but shouldn't we check the sum of noise tasks, instead of just the noisiest ? If we have 20 tasks that contribute 0.9% each, that will not be caught but will still be an issue. 

Reporting the noisiest task makes sense to avoid bloating the metrics, but we should probably take into account the sum:
```
duration_s = df_noise.runtime.values.sum()
```